tests/integration_tests/modules/test_ssh_auth_key_fingerprints.py

```diff
@@ -71,5 +71,5 @@ def test_no_home_directory_created(client: IntegrationInstance):
     assert "system" not in home_output
 
     passwd = client.execute("cat /etc/passwd")
-    assert "nch:" in passwd
-    assert "system:" in passwd
+    assert re.match("nch:", passwd)
+    assert re.match("system:", passwd)
```

Comment:
Thanks for the PR. Though I don't think this is will work for us as re.match will always be None until the full text of passwd exactl matched "nch:" or "system:"

I think your probably want something like the following instead if you are trying to make things strict


```python
assert re.search(r"^nch:", passwd, re.MULTILINE)
...
```

Reply:
On an ubuntu system with LXD installed you can verify the failure with this PR with
```bash
CLOUD_INIT_PLATFORM=lxd_container CLOUD_INIT_CLOUD_INIT_SOURCE=ppa:cloud-init-dev/daily CLOUD_INIT_OS_IMAGE=impish::ubuntu::impish tox -e integration-tests tests/integration_tests/modules/test_ssh_auth_key_fingerprints.py::test_no_home_directory_created
```

Reply:
thanks, @blackboxsw . It looks like you're right. I wasn't expecting a failure, given that Travis CI passed. Looking at the output of some of the jobs, is it because it does not run integration tests?

re Ubuntu with LXD: thank you for the hint! I do encounter an error though, and it looks like I need some additional setup?
```
============================================================================================================================ test session starts =============================================================================================================================
platform linux -- Python 3.8.10, pytest-7.1.1, pluggy-1.0.0 -- /home/jf/cloud-init/.tox/integration-tests/bin/python
cachedir: .tox/integration-tests/.pytest_cache
rootdir: /home/jf/cloud-init, configfile: tox.ini
collected 1 item

tests/integration_tests/modules/test_ssh_auth_key_fingerprints.py::test_no_home_directory_created
------------------------------------------------------------------------------------------------------------------------------- live log setup -------------------------------------------------------------------------------------------------------------------------------
2022-04-05 01:52:28 INFO      integration_testing:clouds.py:79 Detected image: image_id=impish os=ubuntu release=impish
ERROR                                                                                                                                                                                                                                                                  [100%]

=================================================================================================================================== ERRORS ===================================================================================================================================
______________________________________________________________________________________________________________ ERROR at setup of test_no_home_directory_created ______________________________________________________________________________________________________________

    @pytest.fixture(scope="session")
    def session_cloud():
        if integration_settings.PLATFORM not in platforms.keys():
            raise ValueError(
                "{} is an invalid PLATFORM specified in settings. "
                "Must be one of {}".format(
                    integration_settings.PLATFORM, list(platforms.keys())
                )
            )

>       cloud = platforms[integration_settings.PLATFORM]()

tests/integration_tests/conftest.py:105:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/integration_tests/clouds.py:99: in __init__
    self.cloud_instance: BaseCloud = self._get_cloud_instance()
tests/integration_tests/clouds.py:259: in _get_cloud_instance
    return self.pycloudlib_instance_cls(tag=self.instance_tag)
.tox/integration-tests/lib/python3.8/site-packages/pycloudlib/cloud.py:33: in __init__
    self.config = parse_config(config_file)[self._type]
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

config_file = None

    def parse_config(config_file: ConfigFile = None) -> MutableMapping[str, Any]:
        """Find the relevant TOML, load, and return it."""
        possible_configs = []
        if config_file:
            possible_configs.append(config_file)
        if os.environ.get("PYCLOUDLIB_CONFIG"):
            possible_configs.append(Path(os.environ["PYCLOUDLIB_CONFIG"]))
        possible_configs.extend(CONFIG_PATHS)
        for path in possible_configs:
            try:
                config = toml.load(path, _dict=Config)
                log.debug("Loaded configuration from %s", path)
                return config
            except FileNotFoundError:
                continue
            except toml.TomlDecodeError as e:
                raise ValueError(
                    "Could not parse configuration file pointed to by "
                    "{}".format(path)
                ) from e
>       raise ValueError(
            "No configuration file found! Copy pycloudlib.toml.template to "
            "~/.config/pycloudlib.toml or /etc/pycloudlib.toml"
        )
E       ValueError: No configuration file found! Copy pycloudlib.toml.template to ~/.config/pycloudlib.toml or /etc/pycloudlib.toml

.tox/integration-tests/lib/python3.8/site-packages/pycloudlib/config.py:54: ValueError
--------------------------------------------------------------------------------------------------------------------------- Captured stdout setup ----------------------------------------------------------------------------------------------------------------------------
Detected image: image_id=impish os=ubuntu release=impish
----------------------------------------------------------------------------------------------------------------------------- Captured log setup -----------------------------------------------------------------------------------------------------------------------------
2022-04-05 01:52:28 INFO      integration_testing:clouds.py:79 Detected image: image_id=impish os=ubuntu release=impish
============================================================================================================================== warnings summary ==============================================================================================================================
.tox/integration-tests/lib/python3.8/site-packages/_pytest/config/__init__.py:1198
  /home/jf/cloud-init/.tox/integration-tests/lib/python3.8/site-packages/_pytest/config/__init__.py:1198: PytestRemovedIn8Warning: The --strict option is deprecated, use --strict-markers instead.
    self.issue_config_time_warning(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
========================================================================================================================== short test summary info ===========================================================================================================================
ERROR tests/integration_tests/modules/test_ssh_auth_key_fingerprints.py::test_no_home_directory_created - ValueError: No configuration file found! Copy pycloudlib.toml.template to ~/.config/pycloudlib.toml or /etc/pycloudlib.toml
======================================================================================================================== 1 warning, 1 error in 0.65s =========================================================================================================================
ERROR: InvocationError for command /home/jf/cloud-init/.tox/integration-tests/bin/python -m pytest --log-cli-level=INFO -vv tests/integration_tests/modules/test_ssh_auth_key_fingerprints.py::test_no_home_directory_created (exited with code 1)
__________________________________________________________________________________________________________________________________ summary ___________________________________________________________________________________________________________________________________
ERROR:   integration-tests: commands failed
```

Reply:
Ahh @jf right, your local config is lacking a ~/.config/pycloudlib.toml file to tell pycloudlib library how to setup a given cloud platform per this error:
```E       ValueError: No configuration file found! Copy pycloudlib.toml.template to ~/.config/pycloudlib.toml or /etc/pycloudlib.toml
```


 To run LXD tests using pycloudlib, the configuration is setup is simple as it doesn't require additional credentials or a cloud account etc. 

```bash
cat > ~/.config/pycloudlib.toml <<EOF
[lxd]
EOF

# Then you should be able to run the integraiton test I mentioned above and get an altogether different error about nch: not assert failing
CLOUD_INIT_PLATFORM=lxd_container CLOUD_INIT_CLOUD_INIT_SOURCE=ppa:cloud-init-dev/daily CLOUD_INIT_OS_IMAGE=impish::ubuntu::impish tox -e integration-tests tests/integration_tests/modules/test_ssh_auth_key_fingerprints.py::test_no_home_directory_created
```

Reply:
thank you. This gets me further along, but now I am asked for a storage pool:
```
============================================================================================================================== warnings summary ==============================================================================================================================
.tox/integration-tests/lib/python3.8/site-packages/_pytest/config/__init__.py:1198
  /home/jf/cloud-init/.tox/integration-tests/lib/python3.8/site-packages/_pytest/config/__init__.py:1198: PytestRemovedIn8Warning: The --strict option is deprecated, use --strict-markers instead.
    self.issue_config_time_warning(

tests/integration_tests/modules/test_ssh_auth_key_fingerprints.py::test_no_home_directory_created
  /home/jf/cloud-init/.tox/integration-tests/lib/python3.8/site-packages/simplestreams/mirrors/__init__.py:206: DeprecationWarning: The 'warn' method is deprecated, use 'warning' instead
    LOG.warn("got ENOENT for (%s, %s), trying with trailing /",

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
========================================================================================================================== short test summary info ===========================================================================================================================
ERROR tests/integration_tests/modules/test_ssh_auth_key_fingerprints.py::test_no_home_directory_created - RuntimeError: Failure (rc=1): Error: Failed instance creation: No storage pool found. Please create a new storage pool
======================================================================================================================= 2 warnings, 1 error in 13.60s ========================================================================================================================
ERROR: InvocationError for command /home/jf/cloud-init/.tox/integration-tests/bin/python -m pytest --log-cli-level=INFO -vv tests/integration_tests/modules/test_ssh_auth_key_fingerprints.py::test_no_home_directory_created (exited with code 1)
__________________________________________________________________________________________________________________________________ summary ___________________________________________________________________________________________________________________________________
ERROR:   integration-tests: commands failed
```

I think we should be fine now with regards to this specific test (I have added a commit); but I think we do need some docs for contributors on how to run the tests